src/robotFactory.js

```diff
@@ -1,10 +1,78 @@
 'use strict';
 
-class BaseRobot {}
+class BaseRobot {
+  constructor(name, weight, position, chipVersion) {
```

Comment:
```suggestion
  constructor(name, weight, coords, chipVersion) {
```
It would be more meaningful to name this object as it's named in the constructor